src/components/SingleSelect/SingleSelect.jsx

```diff
@@ -0,0 +1,196 @@
+		 */
+		selectedIndex: number,
+		/**
+		 * Object of DropMenu props which are passed thru to the underlying DropMenu component.
+		 */
+		DropMenu: shape(DropMenu.propTypes),
+		/**
+		 * Called when an option is selected.
+		 * Has the signature `(optionIndex, {props, event}) => {}` where `optionIndex` is the new `selectedIndex` or `null`.
+		 */
+		onSelect: func,
+		/**
+		 * *Child Element* - The content rendered in the control when there is no option is selected. Also rendered in the option list to remove current selection.
+		 */
+		Placeholder: any,
+		/**
+		 * *Child Element* - These are menu options. The `optionIndex` is in-order of rendering regardless of group nesting, starting with index `0`.
+		 * Each `Option` may be passed a prop called `isDisabled` to disable selection of that `Option`.
+		 * Any other props pass to Option will be available from the `onSelect` handler.
+		 */
+		Option: any,
+		/**
+		 * *Child Element* - Used to group `Option`s within the menu. Any non-`Option`s passed in will be rendered as a label for the group.
+		 */
+		OptionGroup: any,
+	},
+
+	getDefaultProps() {
+		return {
+			allowReset: true,
+			isDisabled: false,
+			selectedIndex: null,
+			DropMenu: DropMenu.getDefaultProps()
+		};
+	},
+
+	getInitialState() {
+		return {
+			optionGroups: [],
+			flattenedOptionsData: [],
+			ungroupedOptionData: [],
+			optionGroupDataLookup: {}
+		}
+	},
+
+	componentWillMount() {
+		// preprocess the options data before rendering
+		this.setState(DropMenu.preprocessOptionData(this.props, SingleSelect));
+	},
+
+	componentWillReceiveProps(nextProps) {
+		// only preprocess options data when it changes (via new props) - better performance than doing this each render
+		this.setState(DropMenu.preprocessOptionData(nextProps, SingleSelect));
+	},
+
+	render() {
+		const {
+			style,
+			className,
+			allowReset,
+			isDisabled,
+			selectedIndex,
+			onSelect,
+			DropMenu: dropMenuProps
+		} = this.props;
+
+		const {
+			direction,
+			isExpanded
+		} = dropMenuProps;
+
+		const {
+			optionGroups,
+			optionGroupDataLookup,
+			ungroupedOptionData,
+			flattenedOptionsData
+		} = this.state;
+
+		const placeholderProps = _.first(SingleSelect.Placeholder.findInAllAsProps(this.props));
+		const placeholder = _.get(placeholderProps, 'children', 'Select');
+		const isItemSelected = _.isNumber(selectedIndex);
+
+		return (
+			<DropMenu
+				{...dropMenuProps}
+				isDisabled={isDisabled}
+				selectedIndices={isItemSelected ? [selectedIndex] : []}
+				className={boundClassNames('&', className)}
+				onSelect={onSelect}
+				style={style}
+			>
+				<DropMenu.Control>
+					<div className={boundClassNames('&-Control', {
+						'&-Control-is-selected': !isDisabled && isItemSelected,
+						'&-Control-is-expanded': isExpanded,
+						'&-Control-is-disabled': isDisabled
+					})}>
+						<span
+							{...(!isItemSelected ? placeholderProps : null)}
```

Comment:
A little weird to be "spreading" `null`?

`{!isItemSelected ? ...placeholderProps : null}`

Or does that not work in the context of a "tag" in a `render`?